optimum/intel/neural_compressor/trainer.py

```diff
@@ -553,17 +553,18 @@ def compute_loss(self, model, inputs, return_outputs=False):
             self.is_in_train
             and hasattr(self, "agent")
             and hasattr(self.agent, "criterion")
-            and hasattr(self.agent.criterion, "teacher_model_forward")
+            and self.agent.criterion is not None
         ):
             logits = self._get_logits(outputs)
-            teacher_outputs = inputs.pop("teacher_outputs", None)
-            # Compute teacher model outputs
-            if teacher_outputs is None:
-                teacher_outputs = self.agent.criterion.teacher_model_forward(
-                    inputs, teacher_model=self.agent.teacher_model._model
-                )
-
-            teacher_logits = self._get_logits(teacher_outputs)
+            teacher_logits = inputs.pop("teacher_logits", None)
+            if hasattr(self.agent, "on_after_compute_loss"):
+                if teacher_logits is None:
+                    teacher_outputs = self.agent.criterion.teacher_model_forward(inputs)
+                    teacher_logits = self._get_logits(teacher_outputs)
+            elif hasattr(self.agent, "on_post_forward"):

```

Comment:
@PenghuiCheng Currently in [main](https://github.com/huggingface/optimum-intel/blob/main/optimum/intel/neural_compressor/trainer.py#L562) we use the `teacher_model_forward` method to compute the teacher outputs, but I agree that it makes totally sense to use `on_post_forward` to stay compatible for `neural_compressor` v1.12 or under